frontend/src/lib/components/HistoryList/historyListLogic.tsx

```diff
@@ -0,0 +1,139 @@
+    [HistoryActions.CHANGED_KEY_ON_FLAG]: (detail) => `changed the flag key to ${detail.to}`,
+}
+
+function descriptionFrom(historyListItem: HistoryListItem): string | JSX.Element | null {
+    const mapping = actionsMapping[historyListItem.action]
+    return (mapping && mapping(historyListItem.detail)) || null
+}
+
+function humanize(results: HistoryListItem[]): HumanizedHistoryListItem[] {
+    return (results || []).reduce((acc, historyListItem) => {
+        const humanized = descriptionFrom(historyListItem)
+        if (humanized !== null) {
+            acc.push({
+                email: historyListItem.email,
+                name: historyListItem.name,
+                description: humanized,
+                created_at: dayjs(historyListItem.created_at),
+            })
+        }
+        return acc
+    }, [] as HumanizedHistoryListItem[])
+}
+
+/**
+ * Since we may be tracking history for a number of items on the same page
+ * E.g. insights on a dashboard
+ * We have a single logic for all items of a type which caches its responses
+ *
+ * TODO It may be slightly more complex than necessary as paging and filtering are likely to follow soon
+ */
+export const historyListLogic = kea<
+    historyListLogicType<HistoryListItem, HistoryListLogicProps, HumanizedHistoryListItem>
+>({
+    path: ['lib', 'components', 'HistoryList', 'historyList', 'logic'],
+    props: {} as HistoryListLogicProps,
+    key: (props) => `history/${props.type}`,
+    actions: {
+        fetchHistory: () => {},
+        fetchHistorySuccess: (apiResponse: PaginatedResponse<HistoryListItem>) => apiResponse,
+        fetchHistoryFailure: (error: ApiError) => ({ error }),
+    },
+    reducers: ({ props }) => ({
+        isLoading: [
+            false,
+            {
+                fetchHistory: () => true,
+                fetchHistorySuccess: () => false,
+                fetchHistoryFailure: () => false,
+            },
+        ],
+        history: [
+            {} as Record<number, HumanizedHistoryListItem[]>,
+            {
+                fetchHistorySuccess: (state, { results }) => {
+                    const newForId = [...(state[props.id] || []), ...humanize(results)]
+                    return { ...state, [props.id]: newForId }
+                },
+            },
+        ],
+    }),
+    listeners: ({ props, actions }) => ({
+        fetchHistory: async (_, breakpoint) => {
+            let apiResponse: PaginatedResponse<HistoryListItem>
+
+            try {
+                apiResponse = await api.get(`/api/projects/@current/feature_flags/${props.id}/history`)
+                breakpoint()
+                actions.fetchHistorySuccess(apiResponse)
+            } catch (error) {
+                actions.fetchHistoryFailure(error as ApiError)
+                return
+            }
+        },
+    }),
+    events: ({ actions }) => ({
+        afterMount: () => {
+            actions.fetchHistory()
```

Comment:
This will only be called the first time a logic of this `type` is mounted, not when the `id` changes. Rethink if `props` is the best place for `id`